knowledge-base/treelist-get-parent-item-on-insert.md

```diff
@@ -0,0 +1,312 @@
+---
+title: Get the parent item when inserting a child to the TreeList
+description: How to get the parent item when inserting a child to the TreeList
+type: how-to
+page_title: Get the parent item when inserting a child to the TreeList
+slug: treelist-kb-get-parent-item-on-insert
+position: 
+tags: 
+ticketid: 1490309
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>TreeList for Blazor</td>
+		</tr>
+		<tr>
+			<td>Version</td>
+			<td>2.18.0 and later</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+When inserting a child item in the TreeList I would like to be able to get the information for the parent it belongs to and show it to the user.
+
+## Solution
+
+The approaches to handle this would depend of the type of data the component is bound to - [Flat]({%slug treelist-data-binding-flat-data%}) or [Hierarchical]({%slug treelist-data-binding-hierarchical-data%}). The examples below showcase how to handle both scenarios:
+
+* [Get the Parent Item When Inserting a Child to the TreeList When Bound To Flat Data](#get-the-parent-item-when-inserting-a-child-to-the-treelist-when-bound-to-flat-data)
+
+* [Get the Parent Item When Inserting a Child to the TreeList When Bound To Hierarchical Data](#get-the-parent-item-when-inserting-a-child-to-the-treelist-when-bound-to-hierarchical-data)
+
+### Get the Parent Item When Inserting a Child to the TreeList When Bound To Flat Data
+
+#### Step by step Explanations
+
+1. Create a [custom command]({%slug treelist-columns-command%}) to initiate inserting in the `TreeListCommandColumn`.
+1. Pass the `TreeListCommandEventsArgs` object to the `OnClick` handler.
+1. Use the [TreeList state]({%slug treelist-state%}) and its `InsertedItem` field to populate the desired information from the current item that you have in the event arguments.
+
+
+````CSHTML
+@using Telerik.DataSource;
+
+<TelerikTreeList Data="@Data"
+                 EditMode="@TreeListEditMode.Popup"
```

Comment:
Check how things are in InLine mode and whether the ParentItem is needed here too. If so - add it to the snippet.